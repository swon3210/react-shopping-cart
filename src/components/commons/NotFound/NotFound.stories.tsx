import NotFound, { Props } from './NotFound';

export default {
  title: 'Components/Commons/NotFound',
  component: NotFound,
  argTypes: {},
};

const Template = (args: Props) => <NotFound {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  message: '정보를 찾을 수 없습니다.',
};
