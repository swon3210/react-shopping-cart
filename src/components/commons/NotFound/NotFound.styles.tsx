import styled from 'styled-components';

export const NotFound = styled.div`
  width: 100%;
  height: ${({ theme }) => theme.PAGE_HEIGHT};
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  transform: translateY(-50%);
`;

export const Image = styled.img`
  width: 300px;
  margin-bottom: 20px;
`;

export const Message = styled.span`
  font-size: 30px;
  font-weight: bold;
  color: ${({ theme }) => theme.TEXT_COLOR};
`;
