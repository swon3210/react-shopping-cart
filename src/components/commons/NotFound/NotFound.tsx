import * as Styled from './NotFound.styles';
import notFoundPNG from '../../../assets/images/not-found.png';

export interface Props {
  message: string;
}

const NotFound = ({ message }: Props) => {
  return (
    <Styled.NotFound>
      <Styled.Wrapper>
        <Styled.Image src={notFoundPNG} alt="not found image" />
        <Styled.Message>{message}</Styled.Message>
      </Styled.Wrapper>
    </Styled.NotFound>
  );
};

export default NotFound;
