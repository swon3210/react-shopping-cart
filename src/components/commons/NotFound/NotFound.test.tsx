import NotFound from './NotFound';
import { render } from '@testing-library/react';

describe('NotFound Component', () => {
  it('NotFound Snapshot', () => {
    const notFoundUtil = render(<NotFound message="테스트 메세지" />);
    expect(notFoundUtil.asFragment()).toMatchSnapshot();
  });
});
