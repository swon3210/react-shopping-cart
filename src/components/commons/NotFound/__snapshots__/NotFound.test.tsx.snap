// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component NotFound Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa gmpVqK"
  >
    <div
      class="sc-gsDJrp dVRfXu"
    >
      <img
        alt="not found image"
        class="sc-dkPtyc bqPuaS"
        src="not-found.png"
      />
      <span
        class="sc-hKwCoD kCkA-Dq"
      >
        테스트 메세지
      </span>
    </div>
  </div>
</DocumentFragment>
`;
